packages/react-query/src/useQuery.ts

```diff
@@ -7,6 +7,28 @@ import type {
 } from './types'
 import { useBaseQuery } from './useBaseQuery'
 
+export function queryOptions<
+  TQueryFnData = unknown,
+  TError = unknown,
+  TData = TQueryFnData,
+  TQueryKey extends QueryKey = QueryKey,
+>(
+  options: UndefinedInitialDataOptions<TQueryFnData, TError, TData, TQueryKey>,
+): UndefinedInitialDataOptions<TQueryFnData, TError, TData, TQueryKey>
+
+export function queryOptions<
+  TQueryFnData = unknown,
+  TError = unknown,
+  TData = TQueryFnData,
+  TQueryKey extends QueryKey = QueryKey,
+>(
+  options: DefinedInitialDataOptions<TQueryFnData, TError, TData, TQueryKey>,
+): DefinedInitialDataOptions<TQueryFnData, TError, TData, TQueryKey>
+
+export function queryOptions(options: unknown) {
+  return options
+}
```

Comment:
should we move this to the core, @DamianOsipiuk ?

The options are currently based on `UseQueryOptions`, which only exists in react-query. I think each lib has their own impl over `QueryObserverOptions` 🤔

Reply:
Yeah this would not work as in `Vue` we have to wrap every option as a possible `ref`
https://github.com/TanStack/query/blob/alpha/packages/vue-query/src/useQuery.ts#L16

Reply:
okay then let's let every framework export their own `queryOptions` ?

Reply:
Yes, i think that would be necessary.